educational-webapp/src/main/webapp/script.js

```diff
@@ -78,4 +78,121 @@ function alphabetAudio(letter) {
     } else if(letter == 'z') {
         zAudio.play();
     }
-}
\ No newline at end of file
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+  //card options
+  const cardArray = [
+    {
+      name: 'blue',
+      img: 'images/colors/blue.jpg'
+    },
+    {
+      name: 'green',
+      img: 'images/colors/green.jpg'
+    },
+    {
+      name: 'red',
+      img: 'images/colors/red.jpg'
+    },
+    {
+      name: 'yellow',
+      img: 'images/colors/yellow.jpg'
+    },
+    {
+      name: 'red',
+      img: 'images/colors/red.jpg'
+    },
+    {
+      name: 'black',
+      img: 'images/colors/black.jpg'
+    },
+    {
+      name: 'blue',
+      img: 'images/colors/blue.jpg'
+    },
+    {
+      name: 'brown',
+      img: 'images/colors/brown.jpg'
+    },
+    {
+      name: 'yellow',
+      img: 'images/colors/yellow.jpg'
+    },
+    {
+      name: 'brown',
+      img: 'images/colors/brown.jpg'
+    },
+    {
+      name: 'green',
+      img: 'images/colors/green.jpg'
+    },
+    {
+      name: 'black',
+      img: 'images/colors/black.jpg'
+    }
+  ]
+
+  cardArray.sort(() => 0.5 - Math.random())
+
+  const grid = document.querySelector('.grid')
+  const resultDisplay = document.querySelector('#result')
+  var cardsChosen = []
```

Comment:
Does this work to save state? I was under the impression that JavaScript starts fresh on every function call, including re-initializing variables: https://stackoverflow.com/questions/19211768/saving-a-javascript-variable-for-later-usage